backend/backendutil/envelope.go

```diff
@@ -5,11 +5,16 @@ import (
 	"strings"
 
 	"github.com/emersion/go-imap"
+	"github.com/emersion/go-message/charset"
 	"github.com/emersion/go-message/textproto"
 )
 
 func headerAddressList(value string) ([]*imap.Address, error) {
-	addrs, err := mail.ParseAddressList(value)
+	decodedValue, err := charset.DecodeHeader(value)
```

Comment:
It'd be better to use a `mail.AddressParser` instead of double-decoding the header and hope it just works: the decoded value might contain quoted-printable escape sequences.